src/EventService.Data/EventCommentRepository.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using LT.DigitalOffice.EventService.Data.Interfaces;
+using LT.DigitalOffice.EventService.Data.Provider;
+using LT.DigitalOffice.EventService.Models.Db;
+using LT.DigitalOffice.Kernel.Extensions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.EntityFrameworkCore;
+
+namespace LT.DigitalOffice.EventService.Data;
+
+public class EventCommentRepository : IEventCommentRepository
+{
+  private readonly IDataProvider _provider;
+  private readonly IHttpContextAccessor _httpContextAccessor;
+
+  public EventCommentRepository(
+    IDataProvider provider,
+    IHttpContextAccessor httpContextAccessor)
+  {
+    _provider = provider;
+    _httpContextAccessor = httpContextAccessor;
+  }
+
+  public async Task<Guid?> CreateAsync(DbEventComment dbEventComment)
+  {
+    if (dbEventComment is null)
+    {
+      return null;
+    }
+
+    _provider.EventComments.Add(dbEventComment);
```

Comment:
why not async?

Reply:
why? 
"This method is async only to allow special value generators, such as the one used by 'Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo', to access the database asynchronously. For all other cases the non async method should be used."